src/suggestion.c

```diff
@@ -0,0 +1,448 @@
+/*
+ * A series of functions to generate spelling suggestions
+ * 
+ * See suggestion.h for function documentation
+ */
+
+#include <stdlib.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <assert.h>
+#include <unistd.h>
+#include <string.h>
+#include "suggestion.h"
+#include "trie.h"
+
+int has_children(trie_t *t, char *s) {
+    
+    return !trie_search(t, s);
```

Comment:
`trie_search` doesn't return a boolean, so I don't think it's okay to use `!` here.

Reply:
Updated to check return value against NOT_IN_TRIE.